packages/internal/bridge/sdk/src/tokenBridge.ts

```diff
@@ -547,143 +477,640 @@ export class TokenBridge {
 
-    return {
-      feeData: fees,
-      unsignedTx: {
-        data,
-        to: getContractRes.contractAddress,
-        value: txValue,
-        chainId: parseInt(sourceChainId, 10),
-      },
-    };
+    simulations.push({
+      network_id: sourceChainId,
+      estimate_gas: true,
+      simulation_type: 'quick',
+      from: sender,
+      to: this.config.bridgeContracts.rootERC20BridgeFlowRate,
+      input: txData,
+      value: txValue,
+    });
+
+    let axiosResponse:AxiosResponse;
+    const tenderlyAPI = this.getTenderlyEndpoint(sourceChainId);
+    try {
+      axiosResponse = await axios.post(
+        tenderlyAPI,
+        {
+          simulations,
+        },
+        {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      );
+    } catch (error: any) {
+      axiosResponse = error.response;
+    }
+
+    if (axiosResponse.data.error) {
+      throw new BridgeError(
+        `Estimating gas failed with the reason: ${axiosResponse.data.error.message}`,
+        BridgeErrorType.TENDERLY_GAS_ESTIMATE_FAILED,
+      );
+    }
+
+    if (axiosResponse.data.simulation_results.length !== simulations.length) {
+      throw new BridgeError(
+        'Estimating gas failed with mismatched responses',
+        BridgeErrorType.TENDERLY_GAS_ESTIMATE_FAILED,
+      );
+    }
+
+    const tenderlyGasEstimatesRes = {} as DynamicGasEstimatesResponse;
+    const simResults = axiosResponse.data.simulation_results;
+
+    if (simResults.length === 1) {
+      if (simResults[0].simulation.error_message) {
+        throw new BridgeError(
+          `Estimating deposit gas failed with the reason: ${simResults[0].simulation.error_message}`,
+          BridgeErrorType.TENDERLY_GAS_ESTIMATE_FAILED,
+        );
+      } else {
+        tenderlyGasEstimatesRes.sourceChainGas = simResults[0].simulation.gas_used;
+        tenderlyGasEstimatesRes.approvalGas = 0;
+      }
+    } else if (axiosResponse.data.simulation_results.length === 2) {
+      if (simResults[0].simulation.error_message) {
+        throw new BridgeError(
+          `Estimating approval gas failed with the reason: ${simResults[0].simulation.error_message}`,
+          BridgeErrorType.TENDERLY_GAS_ESTIMATE_FAILED,
+        );
+      } else if (simResults[1].simulation.error_message) {
+        throw new BridgeError(
+          `Estimating deposit gas failed with the reason: ${simResults[1].simulation.error_message}`,
+          BridgeErrorType.TENDERLY_GAS_ESTIMATE_FAILED,
+        );
+      } else {
+        tenderlyGasEstimatesRes.approvalGas = simResults[0].simulation.gas_used;
+        tenderlyGasEstimatesRes.sourceChainGas = simResults[1].simulation.gas_used;
+      }
+    } else {
+      throw new BridgeError(
+        `Estimating gas failed with unexpected number responses ${axiosResponse.data.simulation_results.length}`,
+        BridgeErrorType.TENDERLY_GAS_ESTIMATE_FAILED,
+      );
+    }
+    return tenderlyGasEstimatesRes;
```

Comment:
This result error processing could be moved into a `processTenderlySimulations` function